### ISSUE TITLE
Enable/Disable localized layer styles through Catalog tool and Layer Settings

### ISSUE THREAD

**Issue by gitcontributor** [geosolutions-it/MapStore2-C028#103]
For now, all the WMS layers getMap requests are accompanied by ENV param if it was enabled through `localConfig.json`. That causes caching issues if GeoServer caching behavior wasn't setuped correctly for that type of request.

To avoid that issue we need to add the possibility to enable or disable localized layer styles through the Catalog tool and Layer Settings.

*Regarding Catalog tool*:
the checkbox should be placed as per the designs below. If enabled - ENV param should be added to all wms layers presented on the map. Disabled by default.
![screenshot-localhost_8081-2020 04 29-15_15_05](https://user-images.githubusercontent.com/61145641/80594928-c0465f00-89d8-11ea-9903-371b7e714495.png)

*Regarding Layer Settings*:
the checkbox should be placed as per the designs below. It should be in sync with Catalog tool settings but could be customizable for each layer.
![screenshot-localhost_8081-2020 04 29-15_19_15](https://user-images.githubusercontent.com/61145641/80595163-2501b980-89d9-11ea-9817-24184d851753.png)

*Text copy for checkboxes* (to be confirmed): `Enable localized layer styles`.

Seperate requirement - remove ENV param from all background layers requests.

**Comment by gitcontributor** [comment]
@tdipisa 
have two questions: 

**first one**
so am I right regarding logic for the next scenarios:
1) I added WMS layers to the map
2) in the catalog tool I enabled localized styles and env param is presented now across **all** the getMap requesrts
3) I disabled it manually for one of the layers and env param now is missing for **that certain layer**
4) in the catalog tool I disabled localized styles and env param is missing now across **all** the getMap requests
Is that sequence right from a business logic point of view?

_second question_
When localized styles are missing could ENV param be empty string or it should not be presented at all?

**Comment by tdipisa** [comment]
@gitcontributor 
there are some corrections below about the issue description.

>For now, all the WMS layers getMap requests are accompanied by ENV param if it was enabled through localConfig.json.

I didn't know there was this option. Therefore, if this general option if set to false, the new checkbox in Catalog and Layer Settings should be hidden

>the checkbox should be placed as per the designs below. 

After "Show Preview"

>If enabled - ENV param should be added to all wms layers presented on the map. Disabled by default.

If the related config option is checked in the WMS Catalog source configuration UI, then the ENV param is automatically added as query parameter for GetMap requests involving all WMS layers  added to the map from this source. Disabled by default.

>the checkbox should be placed as per the designs below. It should be in sync with Catalog tool settings but could be customizable for each layer.

the checkbox should be placed as per the designs below. As soon as a layer is added to the map, the option in Layer Settings has the same value of the one present in the Catalog source where the layer comes from. Therefore the user is able to change the setting per layer if he wants.

Regarding your questions:

>1) I added WMS layers to the map
>2) in the catalog tool I enabled localized styles and env param is presented now across **all** the getMap requesrts

But only for layers added from the Catalog source where the localized styles option has been enabled

3) I disabled it manually for one of the layers and env param now is missing for **that certain layer**
4) in the catalog tool I disabled localized styles and env param is missing now across **all** the getMap requests

But only for newly layers added from the Catalog source where the localized styles option has been enabled

>When localized styles are missing could ENV param be empty string or it should not be presented at all?

It must not be present at all
